backend/src/main/java/de/neuefische/capstone/backend/services/MyLoansService.java

```diff
@@ -50,4 +51,31 @@ public Loan getLoanDetails(String loanId) {
         throw new NoSuchElementException(userNotFoundExceptionMessage + userId);
 
     }
+
```

Comment:
have you consider using a stream here? 
`if (userData.isPresent()) {
        List<Loan> loans = userData.get().getLoans();
        Optional<Loan> loanToUpdate = loans.stream()
                .filter(loan -> loan.getId().equals(loanId))
                .findFirst();`

Reply:
my problem was to update the element inside the list in the userData object.
i didn't know what to use instead of "set"; and for set i needed the "index"?!

i used "loans.set(index, updatedLoan);" to update the updatedLoan element in the loans List in the userData object and later i called "myLoansRepository.save(userData.get());" to update the userData object (with the updated loans list) in the database.
is there a better way to update an element in a list or to get the index of the element in the list?